sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/LoadBalancerImpl.java

```diff
@@ -248,6 +256,33 @@ protected void beforeCreating() {
             }
         }
 
+        // Reset and update outbound rules
+        List<OutboundRuleInner> innerOutboundRules = innersFromWrappers(this.outboundRules.values());
+        if (null == innerOutboundRules) {
+            innerOutboundRules = new ArrayList<>();
+        }
+        this.innerModel().withOutboundRules(innerOutboundRules);
+
+        for (LoadBalancerOutboundRule outboundRule : this.outboundRules.values()) {
+            // Clear deleted frontend references
+            List<SubResource> refs = outboundRule.innerModel().frontendIpConfigurations();
+            if (refs != null && !refs.isEmpty()) {
+                List<SubResource> existingFrontendIpConfigurations =
+                    refs.stream()
+                        .filter(ref ->
+                            this.frontends().containsKey(ResourceUtils.nameFromResourceId(ref.id()))
+                        )
+                        .collect(Collectors.toList());
+                existingFrontendIpConfigurations = existingFrontendIpConfigurations.isEmpty() ? null : existingFrontendIpConfigurations;
+                outboundRule.innerModel().withFrontendIpConfigurations(existingFrontendIpConfigurations);
+            }
+            // clear deleted backend references
+            SubResource ref = outboundRule.innerModel().backendAddressPool();
+            if(ref != null && !this.backends().containsKey(ResourceUtils.nameFromResourceId(ref.id()))) {
+                outboundRule.innerModel().withBackendAddressPool(null);
+            }
```

Comment:
Is it valid for outbound rule to have null frontend or backend?

Reply:
For this part, I refered to loadBalancer inbound NAT rule, it also has 'clear deleted frontend references' logic, but frontend is mandatory for inbound NAT rule. So from my understanding, this is for clearing purpose before creating outbound rule, when create outbound rule, the frontend and backend will be set. Attach the code for inbound NAT rule part.
```
 // Reset and update inbound NAT rules
  List<InboundNatRuleInner> innerNatRules = innersFromWrappers(this.inboundNatRules.values());
  if (null == innerNatRules) {
      innerNatRules = new ArrayList<>();
  }
  this.innerModel().withInboundNatRules(innerNatRules);

  for (LoadBalancerInboundNatRule natRule : this.inboundNatRules.values()) {
      // Clear deleted frontend references
      SubResource ref = natRule.innerModel().frontendIpConfiguration();
      if (ref != null && !this.frontends().containsKey(ResourceUtils.nameFromResourceId(ref.id()))) {
          natRule.innerModel().withFrontendIpConfiguration(null);
      }
  }
```

Reply:
The question is that when user does not create/update outbound rule, but removed some frontend/backend.

Reply:
If user use below code to update loadbalancer who has outbound rule associate with frontend, the server will delete the frontend, but keep the outbound rule which has no frontend.
```
loadBalancer1
            .update()
            .withoutFrontend(frontendName2)
            .apply();
```